appscale/tools/agents/azure_agent.py

```diff
@@ -446,13 +453,34 @@ def run_instances(self, count, parameters, security_configured, public_ip_needed
     credentials = self.open_connection(parameters)
     subscription_id = str(parameters[self.PARAM_SUBSCRIBER_ID])
     virtual_network = parameters[self.PARAM_GROUP]
+    resource_group = parameters[self.PARAM_RESOURCE_GROUP]
 
     network_client = NetworkManagementClient(credentials, subscription_id)
+    compute_client = ComputeManagementClient(credentials, subscription_id)
+
     subnet = self.create_virtual_network(network_client, parameters,
                                          virtual_network, virtual_network)
 
     active_public_ips, active_private_ips, active_instances = \
       self.describe_instances(parameters)
+
+    # Create an Availability Set for Load Balancer VMs
+    lb_avail_set_name = "lb-availability-set"
+    availability_set_names = [availability_set.name for availability_set in
+                              compute_client.availability_sets.list(resource_group)]
+    if lb_avail_set_name not in availability_set_names:
```

Comment:
`availability_set` needs to be defined if this condition isn't met since it's used below.

Reply:
Good point. I have fixed it in the commit https://github.com/AppScale/appscale-tools/pull/714/commits/09709d4d3ef96e33ce7b40756d04b3d5c02250e0. It doesn't appear to resolve the changes here before the line I changed is different than where you pointed at.